cmd/subctl/deploybroker.go

```diff
@@ -40,31 +42,29 @@ var (
 	defaultComponents = []string{component.ServiceDiscovery, component.Connectivity}
 )
 
+var deployRestConfigProducer = restconfig.NewProducer()
+
 // deployBroker represents the deployBroker command.
 var deployBroker = &cobra.Command{
 	Use:   "deploy-broker",
 	Short: "Deploys the broker",
 	Run: func(cmd *cobra.Command, args []string) {
-		status := cli.NewReporter()
-
-		config, err := restConfigProducer.ForCluster()
-		exit.OnError(status.Error(err, "Error creating REST config"))
-
-		clientProducer, err := client.NewProducerFromRestConfig(config.Config)
-		exit.OnError(status.Error(err, "Error creating client producer"))
-
-		err = deploy.Broker(&deployflags, clientProducer, status)
-		exit.OnError(err)
-
-		err = broker.WriteInfoToFile(config.Config, deployflags.BrokerNamespace, ipsecSubmFile,
-			stringset.New(deployflags.BrokerSpec.Components...), deployflags.BrokerSpec.DefaultCustomDomains, status)
-		exit.OnError(err)
+		exit.OnError(deployRestConfigProducer.RunOnSelectedContext(
+			func(clusterInfo *cluster.Info, namespace string, status reporter.Interface) error {
+				if err := deploy.Broker(&deployflags, clusterInfo.ClientProducer, status); err != nil {
+					return err // nolint:wrapcheck // No need to wrap errors here.
+				}
+
+				return broker.WriteInfoToFile( // nolint:wrapcheck // No need to wrap errors here.
+					clusterInfo.RestConfig, deployflags.BrokerNamespace, ipsecSubmFile,
+					stringset.New(deployflags.BrokerSpec.Components...), deployflags.BrokerSpec.DefaultCustomDomains, status)
+			}, cli.NewReporter()))
```

Comment:
I find it relatively hard to read this anonymous function with such a high level of indentation.
Do you think we can extract it to a named function, for easier code maintainability?